core-web/libs/block-editor/src/lib/extensions/image-tabview-form/image-tabview-form.component.ts

```diff
@@ -0,0 +1,161 @@
+            .pipe(take(1))
+            .subscribe((dotLang) => {
+                this.dotLangs = dotLang;
+                this.search$.next(0);
+            });
+    }
+
+    searchContentlets(offset: number = 0) {
+        if (offset == 0) {
+            this.virtualItems = [];
+        }
+
+        return this.searchService.get(this.params(offset)).pipe(
+            map(({ jsonObjectView: { contentlets } }) => {
+                const items = this.setContentletLanguage(contentlets);
+                this.fillVirtualItems(items);
+                this.loading = false;
+                this.done = !contentlets?.length;
+
+                return this.virtualItems;
+            })
+        );
+    }
+
+    /**
+     * Reset Form Values
+     *
+     * @memberof ImageTabviewFormComponent
+     */
+    resetForm(): void {
+        this.virtualItems = [];
+        this.form.reset({ search: '' }, { emitEvent: false });
+    }
+
+    private params(offset: number): queryEsParams {
+        return {
+            query: ` +catchall:${this.currentSearch}* title:'${this.currentSearch}'^15 +languageId:${this.languageId} +baseType:(4 OR 9) +metadata.contenttype:image/* +deleted:false +working:true`,
+            sortOrder: ESOrderDirection.ASC,
+            limit: 20,
+            offset
+        };
+    }
+
+    /**
+     * This method add the Language to the contentets based on their languageId
+     *
+     * @private
+     * @param {DotCMSContentlet[]} contentlets
+     * @return {*}
+     * @memberof ImageTabviewFormComponent
+     */
+    private setContentletLanguage(contentlets: DotCMSContentlet[]) {
+        return contentlets.map((data) => {
+            return {
+                ...data,
+                language: this.getContentletLanguage(data.languageId)
+            };
+        });
+    }
+
+    private getContentletLanguage(languageId: number): string {
+        const { languageCode, countryCode } = this.dotLangs[languageId];
+
+        if (!languageCode || !countryCode) {
+            return '';
+        }
+
+        return `${languageCode}-${countryCode}`;
+    }
+
+    /**
+     * Create an array of type: [[DotCMSContentlet, DotCMSContentlet], ...]
+     * Due PrimeNg virtual scroll allows only displaying one element at a time [https://primefaces.org/primeng/virtualscroller],
+     * and figma's layout requires displaying two columns of contentlets [https://github.com/dotCMS/core/issues/23235]
+     *
+     * @private
+     * @param {DotCMSContentlet[]} contentlets
+     * @memberof ImageTabviewFormComponent
+     */
+    private fillVirtualItems(contentlets: DotCMSContentlet[]) {
+        contentlets.forEach((contentlet) => {
+            const i = this.virtualItems.length - 1;
+            if (!this.virtualItems[i] || this.virtualItems[i]?.length === 2) {
+                this.virtualItems.push([contentlet]);
+            } else {
+                this.virtualItems[i].push(contentlet);
+            }
+        });
+    }
```

Comment:
An idea:

```JavaScript 
// program to check if the number is even or odd
// take input from the user
const number = prompt("Enter a number: ");

//check if the number is even
if(number % 2 == 0) {
    console.log("The number is even.");
}

// if the number is odd
else {
    console.log("The number is odd.");
}
```

Reply:
I did a refactor. However, I'm checking that the `virtualItem` at position `i` is an array of two `contentlets` (`[DotContentlet, DotContentlet]`). If that is the case, add a new array as a new item.

If I check if the number is even, I going to run into this case:

```js
this.virtualItems[i] = []
this.virtualItems[i] .lenght % 2 == 0 // It's going to be true.
```